File: app/src/main/java/org/strickland/japa/CounterCallback.java
package org.strickland.japa;

/**
 * Callback interface used by CounterService to notify the bound Activity
 * of state changes without going through a broadcast.
 */
public interface CounterCallback {
    void onCountUpdated(int currentBead, int currentRound,
                        int totalBeads, int totalRounds, boolean isComplete);
}

File: app/src/main/kotlin/org/strickland/japa/CounterCallback.kt
package org.strickland.japa

/**
 * Callback interface used by CounterService to notify the bound Activity
 * of state changes without going through a broadcast.
 */
interface CounterCallback {
    fun onCountUpdated(
        currentBead: Int, currentRound: Int,
        totalBeads: Int, totalRounds: Int, isComplete: Boolean
    )
}
